### PR TITLE
Output: Add json output

### PR DESCRIPTION
**Pull request by darox**
This commit adds a command line flag that enables
outputting pwru events in json format.

### CODE REVIEW

File: internal/pwru/output.go [hunk 1]
```diff
@@ -120,7 +181,50 @@ func (o *output) Print(event *Event) {
 	case "arm64":
 		addr = event.Addr
 	}
+	return addr
+}
+
+func getTupleData(event *Event) (tupleData string) {
+	tupleData = fmt.Sprintf("%s:%d->%s:%d(%s)",
```
Comment (brb) [inline internal/pwru/output.go:188]: Could we get this as JSON too?

**Review by brb — COMMENTED** [review]
Thanks! And sorry for taking so long to review.

File: internal/pwru/output.go (hunk 1 above)
Comment (darox) [reply internal/pwru/output.go:188]: I'm not sure what you mean, but running the following command:
`./pwru --output-tuple --output-json`

outputs:
```
{"cpu":0,"func":"tcp_urg","process":"<empty>:(2838)","skb":"0xffff10988cec8f00","tuple":"198.19.248.1:32611->198.19.248.2:2223(tcp)"}
```

jq works on top of it:
```
./pwru --output-tuple --output-json | jq .tuple
"127.0.0.1:51626->127.0.0.1:2379(tcp)"
"127.0.0.1:51626->127.0.0.1:2379(tcp)"
"127.0.0.1:51626->127.0.0.1:2379(tcp)"
"127.0.0.1:2379->127.0.0.1:51626(tcp)"
```

Comment (brb) [reply internal/pwru/output.go:188]: I meant:
```
"tuple": { "sip": ..., "dip": ..., "proto": ..., "sport": ..., "dport": ...}
```

Comment (darox) [reply internal/pwru/output.go:188]: Thank you for explaining. Now I get what you meant. I will implement it.

Comment (darox) [reply internal/pwru/output.go:195]: Implemented, but I think we should refactor and move the pieces of the output to structs. What do you think?


```
{"cpu":2,"func":"__copy_skb_header","process":"containerd-shim:1832","skb":"0xffff1098826d40e0","tuple":{"dip":"1.1.1.1","dport":443,"proto":"tcp","sip":"198.19.249.52","sport":37230}}
```

Comment (brb) [reply internal/pwru/output.go:195]: SGTM!

Comment (darox) [reply internal/pwru/output.go:195]: So do you think we should move it or merge like it is right now?

Comment (brb) [reply internal/pwru/output.go:195]: Refactor and move.

Comment (darox) [reply internal/pwru/output.go:222]: done

**Comment by brb** [comment]
Thanks! Could you drop the second commit (`git fetch -u origin && git rebase origin/main` should do the job instead).

**Comment by brb** [comment]
@darox Thanks. The PR got into a funky state (it's undoing some changes in some previous PRs). Could you rebase against the latest `main` (`git fetch -u origin && git rebase origin/main && git push --force-with-lease`)?

**Comment by darox** [comment]
> @darox Thanks. The PR got into a funky state (it's undoing some changes in some previous PRs). Could you rebase against the latest `main` (`git fetch -u origin && git rebase origin/main && git push --force-with-lease`)?

I'm sorry. I had a merge conflict then fixed it, which merged main into this branch. Then I reset to my commit and pushed. I now executed your command.

**Comment by brb** [comment]
@darox The CI failures look legit. With `--backend=kprobe`, the file content:
```
               SKB    CPU          PROCESS                     FUNC
               SKB    CPU          PROCESS                     FUNC
0xffff91a34c09cae8      8 [/usr/bin/curl:230522]             ip_local_out192.168.18.185:41624->1.0.0.1:8080(tcp)
0xffff91a34c09cae8      8 [/usr/bin/curl:230522]           __ip_local_out192.168.18.185:41624->1.0.0.1:8080(tcp)
0xffff91a34c09cae8      8 [/usr/bin/curl:230522]             nf_hook_slow192.168.18.185:41624->1.0.0.1:8080(tcp)
0xffff91a34c09cae8      8 [/usr/bin/curl:230522] kfree_skb_reason(SKB_DROP_REASON_NETFILTER_DROP)192.168.18.185:41624->1.0.0.1:8080(tcp)
0xffff91a34c09cae8      8 [/usr/bin/curl:230522]   skb_release_head_state192.168.18.185:41624->1.0.0.1:8080(tcp)
0xffff91a34c09cae8      8 [/usr/bin/curl:230522]                tcp_wfree192.168.18.185:41624->1.0.0.1:8080(tcp)
0xffff91a34c09cae8      8 [/usr/bin/curl:230522]         skb_release_data192.168.18.185:41624->1.0.0.1:8080(tcp)
0xffff91a34c09cae8      8 [/usr/bin/curl:230522]             kfree_skbmem192.168.18.185:41624->1.0.0.1:8080(tcp)
```

There is no space between "kfree_skbmem" and "192.168.18.185:41624".

**Comment by darox** [comment]
> @darox The CI failures look legit. With `--backend=kprobe`, the file content:
> 
> ```
>                SKB    CPU          PROCESS                     FUNC
>                SKB    CPU          PROCESS                     FUNC
> 0xffff91a34c09cae8      8 [/usr/bin/curl:230522]             ip_local_out192.168.18.185:41624->1.0.0.1:8080(tcp)
> 0xffff91a34c09cae8      8 [/usr/bin/curl:230522]           __ip_local_out192.168.18.185:41624->1.0.0.1:8080(tcp)
> 0xffff91a34c09cae8      8 [/usr/bin/curl:230522]             nf_hook_slow192.168.18.185:41624->1.0.0.1:8080(tcp)
> 0xffff91a34c09cae8      8 [/usr/bin/curl:230522] kfree_skb_reason(SKB_DROP_REASON_NETFILTER_DROP)192.168.18.185:41624->1.0.0.1:8080(tcp)
> 0xffff91a34c09cae8      8 [/usr/bin/curl:230522]   skb_release_head_state192.168.18.185:41624->1.0.0.1:8080(tcp)
> 0xffff91a34c09cae8      8 [/usr/bin/curl:230522]                tcp_wfree192.168.18.185:41624->1.0.0.1:8080(tcp)
> 0xffff91a34c09cae8      8 [/usr/bin/curl:230522]         skb_release_data192.168.18.185:41624->1.0.0.1:8080(tcp)
> 0xffff91a34c09cae8      8 [/usr/bin/curl:230522]             kfree_skbmem192.168.18.185:41624->1.0.0.1:8080(tcp)
> ```
> 
> There is no space between "kfree_skbmem" and "192.168.18.185:41624".

I will take a look tomorrow.

**Comment by darox** [comment]
@brb all checks passed now.

**Review by brb — APPROVED** [review]
Thanks!

**Pull request merged**